src/components/hint/index.tsx

```diff
@@ -32,127 +34,169 @@ const TARGET_POSITIONS = {
  * @example: https://github.com/wix/react-native-ui-lib/blob/master/demo/src/screens/componentScreens/HintsScreen.js
  * @notes: You can either wrap a component or pass a specific targetFrame
  */
-class Hint extends BaseComponent {
-  static displayName = 'Hint';
 
-  static propTypes = {
-    /**
-     * Control the visibility of the hint
-     */
-    visible: PropTypes.bool,
-    /**
-     * The hint background color
-     */
-    color: PropTypes.string,
-    /**
-     * The hint message
-     */
-    message: PropTypes.oneOfType([PropTypes.string, PropTypes.element]),
-    /**
-     * The hint message custom style
-     */
-    messageStyle: PropTypes.oneOfType([PropTypes.object, PropTypes.number, PropTypes.array]),
-    /**
-     * Icon to show next to the hint's message
-     */
-    icon: PropTypes.oneOfType([PropTypes.object, PropTypes.number]),
-    /**
-     * The icon's style
-     */
-    iconStyle: PropTypes.oneOfType([PropTypes.object, PropTypes.number, PropTypes.array]),
-    /**
-     * The hint's position
-     */
-    position: PropTypes.oneOf(_.values(HINT_POSITIONS)),
-    /**
-     * Provide custom target position instead of wrapping a child
-     */
-    targetFrame: PropTypes.shape({
-      x: PropTypes.number,
-      y: PropTypes.number,
-      width: PropTypes.number,
-      height: PropTypes.number
-    }),
-    /**
-     * Show side tips instead of the middle tip
-     */
-    useSideTip: PropTypes.bool,
-    /**
-     * The hint's border radius
-     */
-    borderRadius: PropTypes.number,
-    /**
-     * Hint margins from screen edges
-     */
-    edgeMargins: PropTypes.number,
-    /**
-     * Hint offset from target
-     */
-    offset: PropTypes.number,
-    /**
-     * Callback for the background press
-     */
-    onBackgroundPress: PropTypes.func,
-    /**
-     * The hint container width
-     */
-    containerWidth: PropTypes.number,
-    /**
-     * The hint's test identifier
-     */
-    testID: PropTypes.string
-  };
+enum HintPositions {
+  TOP = 'top',
+  BOTTOM = 'bottom',
+}
+
+interface HintTargetFrame {
+  x?: number;
+  y?: number;
+  width?: number;
+  height?: number;
+}
+
+interface Position {
+  top?: number;
+  bottom?: number;
+  left?: number;
+  right?: number;
+}
+
+interface HintPositionStyle extends Position {
+  alignItems?: string;
+}
+
+interface Paddings {
```

Comment:
Would make sense, but afaik we'd still need an extra type if we wanted to reuse `NativePaddingKeyType` since it doesn't map to numbers:
```
type NumericValue<T> = {
  [Property in T]: number;
}

type Paddings = NumericValue<NativePaddingKeyType>
```
So seems like the number of types in the file would remain more or less the same 😅. Also, not all of the `NativePaddingKeyType` keys are used here. So I'd prefer keeping things stupid and simple. Unless there's already a mapped type I could reuse (but I couldn't find one)?

Reply:
I see, my mistake :/